modules/videoio/test/test_ffmpeg.cpp

```diff
@@ -235,6 +235,66 @@ const videoio_container_params_t videoio_container_params[] =
 
 INSTANTIATE_TEST_CASE_P(/**/, videoio_container, testing::ValuesIn(videoio_container_params));
 
+typedef tuple<VideoCaptureAPIs, string, int, int, int, int, int> videoio_container_get_params_t;
+typedef testing::TestWithParam<videoio_container_get_params_t > videoio_container_get;
+
+TEST_P(videoio_container_get, read)
+{
+    const VideoCaptureAPIs api = get<0>(GetParam());
+
+    if (!videoio_registry::hasBackend(api))
+        throw SkipTestException("Backend was not found");
+
+    const string fileName = get<1>(GetParam());
+    const int height = get<2>(GetParam());
+    const int width = get<3>(GetParam());
+    const int nFrames = get<4>(GetParam());
+    const int bitrate = get<5>(GetParam());
+    const int fps = get<6>(GetParam());
+
+    VideoCapture container(findDataFile(fileName), api, { CAP_PROP_FORMAT, -1 });
+    if (!container.isOpened())
+        throw SkipTestException("Video stream is not supported");
+
+    const int heightProp = static_cast<int>(container.get(CAP_PROP_FRAME_HEIGHT));
+    ASSERT_EQ(height, heightProp);
+    const int widthProp = static_cast<int>(container.get(CAP_PROP_FRAME_WIDTH));
+    ASSERT_EQ(width, widthProp);
+    const int nFramesProp = static_cast<int>(container.get(CAP_PROP_FRAME_COUNT));
+    ASSERT_EQ(nFrames, nFramesProp);
+    const int bitrateProp = static_cast<int>(container.get(CAP_PROP_BITRATE));
+    ASSERT_EQ(bitrate, bitrateProp);
+    const double fpsProp = container.get(CAP_PROP_FPS);
+    ASSERT_EQ(fps, fpsProp);
+    // remove when PR fixing raw video CAP_PROP_POS_MSEC return value is merged and windows dll is updated
+#ifndef _WIN32
+    vector<int> displayTimeMs;
+    int iFrame = 1;
+    while (container.grab()) {
+        displayTimeMs.push_back(static_cast<int>(container.get(CAP_PROP_POS_MSEC)));
+        const int iFrameProp = static_cast<int>(container.get(CAP_PROP_POS_FRAMES));
+        ASSERT_EQ(iFrame++, iFrameProp);
+    }
+    sort(displayTimeMs.begin(), displayTimeMs.end());
```

Comment:
looks wired. `displayTimeMs` are timestamps from video and they should be sorted by design as you read video sequentially. I propose to remove the sorting.

Reply:
When using `VideoCapture` to demux the packets are not necessarily in display order and as a result the timestamp relates to the [decompression ](https://ffmpeg.org/doxygen/3.1/structAVPacket.html) not presentation time.  The original issue hints at this because the second I-Frame was 246 (file order) and not 250 (display order).

> 242 10.135 False (1, 64)
> 243 10.052 False (1, 62)
> 244 10.093 False (1, 62)
> 245 10.177 False (1, 62)
> 246 10.427 True (1, 3377)
> 247 10.344 False (1, 62)
> 248 10.260 False (1, 62)
> 249 10.302 False (1, 62)
> 250 10.385 False (1, 62)